src/Rules/ListOfRule.php

```diff
@@ -100,12 +100,12 @@ public function processValue($value, Args $args, FieldContext $context): array
 				);
 			} catch (ValueDoesNotMatch | InvalidData $exception) {
 				//TODO - mark key invalid (if not just one higher than previous or if it's first and not 0)
-				$type->addInvalidItem($key, $exception->getInvalidType());
+				$type->addInvalidItem($key, $exception);
 			}
 		}
 
 		if ($type->hasInvalidParameters() || $type->hasInvalidItems()) {
-			throw ValueDoesNotMatch::create($type);
+			throw ValueDoesNotMatch::create($type, $value);
```

Comment:
In case only items are invalid there is no need to track whole value
```
		if (($invalidParams = $type->hasInvalidParameters()) || $type->hasInvalidItems()) {
			throw ValueDoesNotMatch::create(
				$type,
				!$invalidParams ? NoValue::create() : $value,
			);
		}
```

Reply:
//TODO - test case